components/policies/staging/base/konflux-rbac/set-tenant-namespace-visibility/set-tenant-namespace-visibility-clusterpolicy.yaml

```
@@ -0,0 +1,61 @@
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: set-tenant-namespace-visibility
+  annotations:
+    policies.kyverno.io/description: "Sets the 'konflux-ci.dev/visibility' label on tenant namespaces to 'public' if a RoleBinding for 'system:authenticated' exists, and 'private' if it does not."
+spec:
+  background: true
+  rules:
+  - name: set-visibility-public-if-authenticated-rolebinding-exists
+    match:
+      any:
+      - resources:
+          kinds:
+          - Namespace
+          selector:
+            matchLabels:
+              konflux-ci.dev/type: tenant
+    context:
+      - name: checkAuthenticatedRoleBinding
+        apiCall:
+          urlPath: "/apis/rbac.authorization.k8s.io/v1/namespaces/{{request.object.metadata.name}}/rolebindings"
+          method: GET
+          jmesPath: "items[?subjects[?name=='system:authenticated' && kind=='Group']]"
+    preconditions:
+      all:
+      - key: "{{ checkAuthenticatedRoleBinding | length(@) }}"
+        operator: GreaterThanOrEquals
+        value: 1 
+    mutate:
+      patchStrategicMerge:
+        metadata:
+          labels:
+            konflux-ci.dev/visibility: public
```

I'm not sure about this, but it would be nice if we could condense these rules into a single one.
Maybe something like the following?

```suggestion
    context:
      - name: visibility
        apiCall:
          urlPath: "/apis/rbac.authorization.k8s.io/v1/namespaces/{{request.object.metadata.name}}/rolebindings"
          method: GET
          jmesPath: "items[?subjects[?name=='system:authenticated' && kind=='Group']] | length(@) == 0 && 'private' || 'public'"
    mutate:
      patchStrategicMerge:
        metadata:
          labels:
            konflux-ci.dev/visibility: {{ visibility }}
```